src/libraries/System.Private.CoreLib/src/System/HashCode.cs

```diff
@@ -305,6 +307,33 @@ public void Add<T>(T value, IEqualityComparer<T>? comparer)
             Add(value is null ? 0 : (comparer?.GetHashCode(value) ?? value.GetHashCode()));
         }
 
+        /// <summary>Adds a span of bytes to the hash code.</summary>
+        /// <param name="value">The span.</param>
+        /// <remarks>
+        /// While the implementation guarantees that the added span will result in the same
+        /// behavior within the same process, it does not guarantee that the result would match
+        /// instead adding each byte value individually.
+        /// </remarks>
+        public void Add(ReadOnlySpan<byte> value)
+        {
+            ref byte pos = ref MemoryMarshal.GetReference(value);
+            ref byte end = ref Unsafe.Add(ref pos, value.Length);
+
+            // Add four bytes at a time until the input has fewer than four bytes remaining.
+            while ((int)Unsafe.ByteOffset(ref pos, ref end) >= sizeof(int))
```

Comment:
The pattern `(int)Unsafe.ByteOffset(...)` incurs overflow checks on x64. Use `(nint)Unsafe.ByteOffset(...)` instead to avoid them.